index_settings_test.go

```diff
@@ -858,6 +903,59 @@ func TestIndex_ResetSortableAttributes(t *testing.T) {
 	}
 }
 
+func TestIndex_ResetTypoTolerance(t *testing.T) {
+	type args struct {
+		UID    string
+		client *Client
+	}
+	tests := []struct {
+		name     string
+		args     args
+		wantTask *Task
+		wantResp *TypoTolerance
+	}{
+		{
+			name: "TestIndexBasicResetTypoTolerance",
+			args: args{
+				UID:    "indexUID",
+				client: defaultClient,
+			},
+			wantTask: &Task{
+				UID: 1,
+			},
+			wantResp: &defaultTypoTolerance,
+		},
+		{
+			name: "TestIndexResetTypoToleranceWithCustomClient",
+			args: args{
+				UID:    "indexUID",
+				client: customClient,
```

Comment:
Do you see a good value in keeping two same tests just changing the client? Since they implement the same `get`, `post`, `put`, `patch`, and `delete` HTTP methods, we can trust they will be ok in every test.

Think about this, and maybe we could remove the `customClient` testing from most of the tests. This will simplify a lot of the test suite without losing trustability.

Reply:
No, I think you are right, keep a test for the `customClient` in the client test should be enough. I will open an issue about it.